pkg/utils/utils.go

```diff
@@ -258,3 +292,124 @@ func CleanCachedNetConf(cRefPath string) error {
+		return fmt.Errorf("failed to write the sender protocol address in the ARP packet: %v", writeErr)
+	}
+	_, _ = arpPacket.Write([]byte{0xff, 0xff, 0xff, 0xff, 0xff, 0xff}) // Target hardware address is the Broadcast MAC.
+	if _, writeErr := arpPacket.Write(srcIP.To4()); writeErr != nil {  // Target protocol address
+		return fmt.Errorf("failed to write the target protocol address in the ARP packet: %v", writeErr)
+	}
+
+	sockAddr := syscall.SockaddrLinklayer{
+		Protocol: htons(syscall.ETH_P_ARP),                                // Ethertype of ARP (0x0806)
+		Ifindex:  iface.Index,                                             // Interface Index
+		Hatype:   1,                                                       // Hardware Type: 1 is Ethernet
+		Pkttype:  0,                                                       // Packet Type.
+		Halen:    6,                                                       // Hardware address Length: 6 bytes for MAC address
+		Addr:     [8]byte{0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff}, // Address is the broadcast MAC address.
+	}
+
+	// Create a socket such that the Ethernet header would constructed by the OS. The arpPacket only contains the ARP payload.
+	soc, err := syscall.Socket(syscall.AF_PACKET, syscall.SOCK_DGRAM, int(htons(syscall.ETH_P_ARP)))
+	if err != nil {
+		return fmt.Errorf("failed to create AF_PACKET datagram socket: %v", err)
+	}
+	defer syscall.Close(soc)
+
+	if err := syscall.Sendto(soc, arpPacket.Bytes(), 0, &sockAddr); err != nil {
+		return fmt.Errorf("failed to send Gratuitous ARP for IPv4 %s on Interface %s: %v", srcIP.String(), iface.Name, err)
+	}
+
+	return nil
+}
+
+// SendUnsolicitedNeighborAdvertisement sends an unsolicited neighbor advertisement packet with the provided source IP over the provided interface.
+func SendUnsolicitedNeighborAdvertisement(srcIP net.IP, iface net.Interface) error {
+	/* As per RFC 4861, a link-layer address change can multicast a few unsolicited neighbor advertisements to all nodes to quickly
+	 * update the cached link-layer addresses that have become invalid. In the case of SRIOV-CNI, an address can be reused for
+	 * different pods. Each pod could likely have a different link-layer address in this scenario, which makes the Neighbor Cache
+	 * entries residing in the neighbors to be an invalid. The unsolicited neighbor advertisement should update the link-layer address
+	 * accordingly for the IPv6 entry.
+	 * However if any of these conditions are true:
+	 *  - The IPv6 address was not reused for the new pod.
+	 *  - No prior established communication with the neighbor.
+	 * Then the neighbor receiving this unsolicited neighbor advertisement would be silently discard. This behavior is described
+	 * in RFC 4861 section 7.2.5. This is acceptable behavior since the purpose of sending an unsolicited neighbor advertisement
+	 * is not to create a new entry but rather update already existing invalid entries.
```

Comment:
If you send unsolicited NA, then this message changes existing entry to STALE as RFC Section 7.2.5's following statement. Does it make sense? 

```
   - If the advertisement's Solicited flag is set, the state of the
     entry is set to REACHABLE; otherwise, it is set to STALE.
```

Reply:
In [7.2.6](https://www.rfc-editor.org/rfc/rfc4861.html#section-7.2.6).  Sending Unsolicited Neighbor Advertisements:

"In either case, neighboring nodes will immediately change the state of their Neighbor Cache entries for the Target Address to STALE, prompting them to verify the path for reachability. "

I believe IPv6 is special in a way that it tests connectivity both ways, that's why it needs to verify the path for reachability.

Reply:
So, regarding ipv6, this change is to invalidate current existing entry from the node. Is it right?

Reply:
Yes that's the most relevant aspect of this change where there could be a reuse of an IP address with a different pod (different MAC address on the container interface).

Reply:
I hope I was able to answer your concerns.